build.gradle

```diff
@@ -46,7 +46,12 @@ subprojects {
     afterEvaluate {
         jar {
             manifest {
-                attributes  'Automatic-Module-Name': moduleName
+                attributes  'Automatic-Module-Name': moduleName,
+                    "Implementation-Version": archiveVersion,
+                    "Implementation-Title": "PerfMark",
+                    "Implementation-Vendor": "Carl Mastrangelo",
+                    "Implementation-URL": "https://www.perfmark.io/",
+                    "Carl-Is-Awesome": "true"
```

Comment:
:smile:

Reply:
It's the little things that make working on this fun.  😄


(I looked through the gRPC one to see what fields were set, there's surprisingly little guidance on what attributes *should* be in here. )

Reply:
Yeah, everything is so generic and abstract it isn't clear what is used by what and to which purposes. You have to read between the lines. You'd sort of imagine [the java tutorial](https://docs.oracle.com/javase/tutorial/deployment/jar/packageman.html) would have more words talking about what is going on, but its definitions of the fields leads something to be desired... You can look at the [actual specification](https://docs.oracle.com/javase/8/docs/technotes/guides/jar/jar.html) which uses _even more words_ to say absolutely nothing about the meaning of the fields.

nmittler added the manifest to gRPC, but at my suggestion after having used it on my previous project: https://github.com/googlegsa/library/commit/b0bbccdd51ec71cdf595561dbad2622e2f392fe7 . I just sort of accept it is the blind leading the blind.

I'll note that I think we are all "doing it wrong" as we aren't putting the attributes within a section for the package. If you look at that Java tutorial you can see `Name` which creates sections. The `Name` doesn't matter for standalone jars much containing a single package, but if we used a name section then any uber jars would have a still-working manifest. But even the [gradle documentation](https://docs.gradle.org/current/userguide/building_java_projects.html#sec:jar_manifest) doesn't describe how to do that. I _think_ we would use [this overload of attributes](https://docs.gradle.org/current/javadoc/org/gradle/api/java/archives/Manifest.html#attributes-java.util.Map-java.lang.String-) to do so. Maybe this is why I see grpc version numbers of `0` still in the wild. Although with modules that should matter less...